### PR TITLE
[FIX] hr_attendance_analysis various bugs

### PR DESCRIPTION
**Pull request by tafaRU**
This PR contains the changes made on https://code.launchpad.net/~agilebg/hr-timesheet/fix-hr_attendance_analysis-various-bugs before the project was hosted on github plus some others.
## Main changes
1. Merged branch lp:~agilebg/hr-timesheet/fix-hr_attendance_analysis-reference-is-not-defined and moved the related MP https://code.launchpad.net/~agilebg/hr-timesheet/fix-hr_attendance_analysis-reference-is-not-defined/+merge/221250
2. Merged branch lp:~agilebg/hr-timesheet/fix-hr_attendance_analysis-bracket and moved the related MP https://code.launchpad.net/~agilebg/hr-timesheet/fix-hr_attendance_analysis-bracket/+merge/221278
3. Merged branch lp:~agilebg/hr-timesheet/fix-hr_attendance_analysis-report-timezone and moved the related MP https://code.launchpad.net/~agilebg/hr-timesheet/fix-hr_attendance_analysis-report-timezone/+merge/221374
4. Merged branch lp:~agilebg/hr-timesheet/hr_attendance_analysis-imp-time_difference-message and moved the related MP https://code.launchpad.net/~agilebg/hr-timesheet/hr_attendance_analysis-imp-time_difference-message/+merge/225275
5. Deleted some unused methods and applied a bit of refactoring
6. Fixed the total leaves calculation: now the current_total_leaves value is calculated considering the working time and the timezone
7. Added hr_attendance_analysis/test/attendances.yml

Thank you.

### CODE REVIEW

**Comment by eLBati** [comment]
Thanks @tafaRU , could you also point out the reviews already done on the old MPs?

**Comment by tafaRU** [comment]
@eLBati I've just updated the description including the related MPs moved.

File: hr_attendance_analysis/wizard/print_calendar_report.py [hunk 1]
```diff
@@ -200,17 +233,58 @@ def print_calendar(self, cr, uid, ids, context=None):
...
                     # if ends after today
                     if date_to > current_date_end:
                         date_to = current_date_end
-                    current_total_leaves = attendance_pool.time_sum(
-                        current_total_leaves,
-                        (date_to - date_from).total_seconds() / 60.0 / 60.0)
+                    date_from = date_from.replace(
+                        tzinfo=pytz.utc).astimezone(active_tz)
+                    date_to = date_to.replace(
+                        tzinfo=pytz.utc).astimezone(active_tz)
```
Comment (eLBati) [inline hr_attendance_analysis/wizard/print_calendar_report.py:268]: Why conversion to user timezone?
Is `matched_schedule` aware of timezone?

Comment (tafaRU) [inline hr_attendance_analysis/wizard/print_calendar_report.py:268]: date_from and date_to are not aware of timezone so I thought to convert them as we made for the attendances.

File: hr_attendance_analysis/hr_attendance.py [hunk 2]
```diff
@@ -293,11 +316,13 @@ def _get_attendance_duration(self, cr, uid, ids, field_name, arg, context=None):
...
+                                    cr, uid, matched_schedule_ids[0],
+                                    context=context
+                                )
                                 attendance_start_hour = (
-                                    attendance_start.hour + attendance_start.minute / 60.0
-                                    + attendance_start.second / 60.0 / 60.0
-                                    )
+                                    self.datetime_to_hour(attendance_start)
+                                )
                                 if attendance_start_hour >= (
```
Comment (eLBati) [inline hr_attendance_analysis/hr_attendance.py]: Why the fix from https://code.launchpad.net/~agilebg/hr-timesheet/fix-hr_attendance_analysis-bracket/+merge/221278 is missing here and at line 343?

File: hr_attendance_analysis/wizard/print_calendar_report.py (hunk 1 above)
Comment (eLBati) [inline hr_attendance_analysis/wizard/print_calendar_report.py:268]: After investiganting, I saw the working schedule is not converted to timezone. That is, if your working schedule is from 09:00 to 13:00 (as you see it in the web interface), in the DB the values are exactly 9 and 13.
On the other hand, the leaves (as well as the attendances) are always saved as UTC in the DB. That is, if you leave at 09:00 and your timezone is GTM+2, the DB will contain 07:00.
So, the conversion here is correct
:+1:

**Comment by pedrobaeza** [comment]
Please rebase to have last PEP8 changes included. You might have to resolve conflicts.

**Comment by eLBati** [comment]
@tafaRU rebased at https://github.com/tafaRU/hr-timesheet/pull/2

**Comment by pedrobaeza** [comment]
There are some PEP8 errors introduced in this PR.

**Comment by tafaRU** [comment]
@pedrobaeza I fixed the PEP8 errors and the test/attendances.yml.
Now the Travis CI light is green colored :grin:

**Comment by rvalyi** [comment]
:+1:  LGTM

**Comment by bwrsandman** [comment]
:+1:

**Comment by pedrobaeza** [comment]
Thank you very much.

:+1: 

Regards.

**Pull request merged**